src/sema/variables.rs

```diff
@@ -233,9 +235,22 @@ pub fn variable_decl<'a>(
 
                 visibility = Some(v.clone());
             }
+            pt::VariableAttribute::StorageType(s) => {
+                storage_type = Some(s.clone());
+            }
         }
     }
 
+    if ns.target == Target::Soroban && storage_type.is_none() {
+        ns.diagnostics.push(Diagnostic::warning(
+            def.loc,
+            format!(
+                "storage type not specified for `{}`, defaulting to `persistent`",
+                def.name.as_ref().unwrap().name
+            ),
+        ));
+    }
```

Comment:
If the `storage_type` is specified and the target is not Soroban, we should give an error and not silently ignore.

Reply:
Would be nice to have test cases for these